once_test.py

```diff
@@ -792,21 +789,36 @@ async def async_yielding_iterator():
         self.assertEqual([i async for i in async_yielding_iterator()], [1, 2, 3])
         self.assertEqual([i async for i in async_yielding_iterator()], [1, 2, 3])
 
-    @unittest.skip("This currently hangs and needs to be fixed, GitHub Issue #12")
     async def test_failing_generator(self):
         counter = Counter()
 
         @once.once
         async def sample_failing_fn():
             yield counter.get_incremented()
-            raise ValueError("expected failure")
+            result = counter.get_incremented()
+            yield result
+            if result == 2:
+                raise ValueError("expected failure after 2.")
```

Comment:
I think "expected" is confusing here. Maybe just `raise ValueError("2 is an ugly number")` or more professionally `raise ValueError("we raise an error when result is exactly 2")`

Reply:
done